internal/abci/abci.go

```diff
@@ -96,7 +96,11 @@ type AbciApp struct {
 	txApp TxApp
 
 	consensusParams *chain.ConsensusParams
-	chainContext    *common.ChainContext
+	// As the consensus params are persisted before calling NotifyHeight, we need to track
+	// the last blocks migration status to correctly persist the migration status in the chain state.
+	lastMigrationStatus types.MigrationStatus
```

Comment:
This seems wrong. `ChainContext`, directly below this, as a `NetworkParameters`, which has this exact same variable. Furthermore, there is a standard for how we handle network parameters; everything else gets persisted in `StoreDiff`. I don't see why this would be any different. We should handle the tracking of the last migration status in the same way that we handle all other update-able network variables.